drivers/ethernet/eth_xlnx_gem.c

```diff
@@ -1449,12 +1449,26 @@ static void eth_xlnx_gem_handle_rx_pending(const struct device *dev)
 		reg_val = sys_read32(reg_ctrl);
```

Comment:
Maybe we should use another variable for reg_ctrl, so that reg_val can be used in the if block below?

Reply:
The reg_val variable is used twice sequentially in order to evaluate the contents of two different memory locations. First, the /address/ word of the current descriptor in the RX buffer descriptor ring is read and checked for the presence of the 'Used' bit. Afterwards, reg_val is re-used as storage for the contents of the current descriptor's /control/ word, which /should/ contain the Start-Of-Frame bit. So line 1463 isn't a re-read, but a read of another memory location compared to line 1449, where the local variable the data is being read to is being re-used.

Reply:
Yeah, I know.  What I meant was you read the location of `reg_addr` twice. 1st at line 1441 and 2nd at 1465 because you overwrite the variable with `reg_val = sys_read32(reg_ctrl)` at 1449.

But, I assume it's a minor issue and I approve this PR anyway.